src/crdt.rs

```diff
@@ -583,14 +622,43 @@ impl Crdt {
     /// * `from` - identity of the sender of the updates
     /// * `update_index` - the number of updates that `from` has completed and this set of `data` represents
     /// * `data` - the update data
-    fn apply_updates(&mut self, from: PublicKey, update_index: u64, data: &[ReplicatedData]) {
+    fn apply_updates(
+        &mut self,
+        from: PublicKey,
+        update_index: u64,
+        data: &[ReplicatedData],
+        external_liveness: &[(PublicKey, u64)],
+    ) {
         trace!("got updates {}", data.len());
         // TODO we need to punish/spam resist here
         // sig verify the whole update and slash anyone who sends a bad update
         for v in data {
             self.insert(&v);
         }
+
+        for (pk, external_remote_index) in external_liveness.iter() {
```

Comment:
It's more idiomatic to write `&external_liveness` than `external_liveness.iter()`. The tool Clippy will give you a bunch of good tips like that.

Reply:
Super helpful, thanks!